llm/server.go

```diff
@@ -899,7 +899,15 @@ func (s *llmServer) Detokenize(ctx context.Context, tokens []int) (string, error
 func (s *llmServer) Close() error {
 	if s.cmd != nil {
 		slog.Debug("stopping llama server")
-		return s.cmd.Process.Kill()
+		if err := s.cmd.Process.Kill(); err != nil {
+			return err
+		}
+
+		if err := s.cmd.Wait(); err != nil {
```

Comment:
Quick update.  I believe the existing close logic in sched.go should be sufficient, so the only thing I think we should do is remove the extra go routine as noted above to avoid duplicate waits.